tesseract/tesseract_planning/tesseract_motion_planners/src/hybrid/descartes_trajopt_array_planner.cpp

```diff
@@ -0,0 +1,107 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <tesseract_motion_planners/hybrid/descartes_trajopt_array_planner.h>
+
+namespace tesseract_motion_planners
+{
+
+template<typename FloatType>
+DescartesTrajOptArrayPlanner<FloatType>::DescartesTrajOptArrayPlanner(std::string name) : MotionPlanner(std::move(name)), status_category_(std::make_shared<tesseract_common::GeneralStatusCategory>(name))
+{
+}
+
+template<typename FloatType>
+bool DescartesTrajOptArrayPlanner<FloatType>::terminate()
+{
+  CONSOLE_BRIDGE_logWarn("Termination of ongoing optimization is not implemented yet");
+  return false;
+}
+
+template<typename FloatType>
+void DescartesTrajOptArrayPlanner<FloatType>::clear()
+{
+  request_ = PlannerRequest();
+  trajopt_planner_.clear();
+  descartes_planner_.clear();
+}
+
+template<typename FloatType>
+tesseract_common::StatusCode DescartesTrajOptArrayPlanner<FloatType>::isConfigured() const
+{
+  tesseract_common::StatusCode descartes_status = descartes_planner_.isConfigured();
+  tesseract_common::StatusCode trajopt_status = trajopt_planner_.isConfigured();
+
+  if (!descartes_status && !trajopt_status)
+    return tesseract_common::StatusCode(status_category_->IsNotConfigured, status_category_);
+
+  if (!descartes_status)
+    return descartes_status;
+
+  if (!trajopt_status)
+    return trajopt_status;
+
+  return tesseract_common::StatusCode(status_category_->IsConfigured, status_category_);
+}
+
+template<typename FloatType>
+tesseract_common::StatusCode DescartesTrajOptArrayPlanner<FloatType>::solve(PlannerResponse& response)
+{
+  tesseract_common::StatusCode config_status = isConfigured();
+  if (!config_status)
+  {
+    response.status = config_status;
+    CONSOLE_BRIDGE_logError("Planner %s is not configured", name_.c_str());
+    return config_status;
+  }
+
+  //Solve problem using descartes. Results are stored in the response
+  tesseract_motion_planners::PlannerResponse descartes_planning_response;
+  tesseract_common::StatusCode descartes_status = descartes_planner_.solve(descartes_planning_response);
+  if (!descartes_status && (descartes_status.value() != DescartesMotionPlannerStatusCategory::FoundValidSolutionInCollision))
+  {
+    response = std::move(descartes_planning_response);
+    return descartes_status;
+  }
+
+  tesseract_motion_planners::PlannerResponse trajopt_planning_response;
+  tesseract_common::StatusCode trajopt_status = trajopt_planner_.solve(trajopt_planning_response);
```

Comment:
It would seem that the trajopt config needs to be updated with the descartes trajectory as seed_trajectory_. 

Also, does the trajopt_planning_response not need to be copied into response as well. It seems that you are returning the descartes trajectory.

Reply:
Good catch! I fixed it.

Reply:
The latest rebase addresses this.